### ISSUE TITLE
`test_wheel.py::test_local_lib` failure in CI

### ISSUE THREAD

**Issue by rgommers** [mesonbuild/meson-python#93]
Test failure just observed in https://github.com/FFY00/meson-python/runs/7202624920?check_suite_focus=true. This should be due to the Meson 0.63.0 release on July 3rd (2 days ago):

```
________________________________ test_local_lib ________________________________
virtual_env = PosixPath('/tmp/mesonpy-test-venv-n30nt69r/bin/python')
wheel_link_against_local_lib = PosixPath('/tmp/pytest-of-runner/pytest-0/test0/mesonpy-test-1jnqmmle/link_against_local_lib-1.0.0-py3-none-linux_x86_64.whl')
    @pytest.mark.skipif(platform.system() != 'Linux', reason='Unsupported on this platform for now')
    def test_local_lib(virtual_env, wheel_link_against_local_lib):
        subprocess.check_call([virtual_env, '-m', 'pip', 'install', wheel_link_against_local_lib])
>       subprocess.check_output([
            virtual_env, '-c', 'import example; print(example.example_sum(1, 2))'
        ]).decode() == '3'
virtual_env = PosixPath('/tmp/mesonpy-test-venv-n30nt69r/bin/python')
wheel_link_against_local_lib = PosixPath('/tmp/pytest-of-runner/pytest-0/test0/mesonpy-test-1jnqmmle/link_against_local_lib-1.0.0-py3-none-linux_x86_64.whl')
tests/test_wheel.py:119: 
...
---------------------------- Captured stdout setup -----------------------------
! Using Meson to generate the project metadata (no `project` section in pyproject.toml)
+ meson setup --native-file=/home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-native-file.ini -Ddebug=false -Dstrip=true -Doptimization=2 --prefix=/opt/hostedtoolcache/Python/3.11.0-beta.3/x64 /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/build
The Meson build system
Version: 0.63.0
Source dir: /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib
Build dir: /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/build
Build type: native build
Project name: link-against-local-lib
Project version: 1.0.0
C compiler for the host machine: cc (gcc 9.4.0 "cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0")
C linker for the host machine: cc ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency python3 found: YES 3.8
Program python3 found: YES (/home/runner/work/meson-python/meson-python/.nox/test-3-11/bin/python)
Build targets in project: 2
link-against-local-lib 1.0.0
  User defined options
    Native files: /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-native-file.ini
    debug       : false
    optimization: 2
    prefix      : /opt/hostedtoolcache/Python/3.11.0-beta.3/x64
    strip       : true
Found ninja-1.10.2.git.kitware.jobserver-1 at /home/runner/work/meson-python/meson-python/.nox/test-3-11/bin/ninja
+ meson compile
[1/5] Compiling C object libexample.so.p/examplelib.c.o
[2/5] Linking target libexample.so
[3/5] Compiling C object example.cpython-311-x86_64-linux-gnu.so.p/examplemod.c.o
[4/5] Generating symbol file libexample.so.p/libexample.so.symbols
[5/5] Linking target example.cpython-311-x86_64-linux-gnu.so
+ meson install --destdir /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/install
ninja: Entering directory `/home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/build'
ninja: no work to do.
Installing libexample.so to /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/install/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/x86_64-linux-gnu
Stripping target 'libexample.so'.
Installing example.cpython-311-x86_64-linux-gnu.so to /home/runner/work/meson-python/meson-python/tests/packages/link-against-local-lib/.mesonpy-7s3pci72/install/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/site-packages
Stripping target 'example.cpython-311-x86_64-linux-gnu.so'.
WARNING Using heuristics to map files to wheel, this may result in incorrect locations
Copying files to wheel...
[0/1] example.cpython-311-x86_64-linux-gnu.so
[1/1] .link_against_local_lib.mesonpy.libs/libexample.so
----------------------------- Captured stdout call -----------------------------
Processing /tmp/pytest-of-runner/pytest-0/test0/mesonpy-test-1jnqmmle/link_against_local_lib-1.0.0-py3-none-linux_x86_64.whl
Installing collected packages: link-against-local-lib
Successfully installed link-against-local-lib-1.0.0
----------------------------- Captured stderr call -----------------------------
WARNING: You are using pip version 22.0.4; however, version 22.1.2 is available.
You should consider upgrading via the '/tmp/mesonpy-test-venv-n30nt69r/bin/python -m pip install --upgrade pip' command.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: dynamic module does not define module export function (PyInit_example)
----------- coverage: platform linux, python 3.11.0-beta-3 -----------
Coverage HTML written to dir /home/runner/work/meson-python/meson-python/.nox/test-3-11/htmlcov
Coverage XML written to file /home/runner/work/meson-python/meson-python/.nox/test-3-11/coverage-3.11.xml
=========================== short test summary info ============================
FAILED tests/test_wheel.py::test_local_lib - subprocess.CalledProcessError: C...
```

This is due to https://github.com/mesonbuild/meson/pull/10369, because it only occurs for Python >= 3.9 with a missing `PyInit_*` symbol. The puzzling thing is that it only fails for this one test, and that `PyMODINIT_FUNC` is used correctly. While in the one other test case that has it, it looks like there's a bug. And I cannot reproduce it locally yet.

**Comment by FFY00** [comment]
I triggered a re-rerun anyway.

**Comment by dnicolodi** [comment]
This was due to some issues with the test projects definitions. Has been fixed in #198.
